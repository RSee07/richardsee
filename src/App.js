import React from 'react'
import { BrowserRouter } from 'react-router-dom'
import styled, { injectGlobal } from 'styled-components'
import ScrollToTop from './components/ScrollToTop'
import personal from './static/data/personal.json'
import Content from './components/pages/Content'
import { color } from './components/Helper'

// Inject styles directly. Cannot apply styled component to body so inject directly
injectGlobal`
    body {
        height: 100%;
        width: 100%;
        margin: 0;
        padding: 0;
        font-family: sans-serif;
        /* Color most closely matches the color where the clipping on iOS occurs */
        background: ${color.backgroundFallback}; /* Fallback for old browsers */

        &:after{
            content: "";
            position: fixed; /* Stretch a fixed position to the whole screen */
            top: 0;
            height: 100vh; /* Fix for mobile browser address bar appearing disappearing */
            left: 0;
            right: 0;
            z-index: -1; /* Needed to keep in the background */
            background: ${color.background};
            background-size: 100% 100%;
        }
    }
`

// Forces gradient background to be over the mobile viewport only
// Without this on iOS gradient over the entire height of the page
// Having this works for Android. iOS has the bottom pop
// Probably not needed for desktop
const GradientBackground = styled.div`
    width: 100vw;
    min-height: 100vh;
    position: fixed;
    top: 0;
    bottom: 0;
    background: ${color.background};
    z-index: -10;
    background-position: top center;
    background-size: cover;
    background-repeat: no-repeat;
    background-attachment: scroll;
`

export default class App extends React.Component {
    // State definition
    state = {
        data: ""
    }

    componentDidMount() {
        // Set data to JSON data once component has mounted
        this.setState({
            data: personal
        })
    }

    render() {
        return (
            <BrowserRouter>
                <ScrollToTop>
                    {this.state.data && <Content data={this.state.data} />}
                </ScrollToTop>
            </BrowserRouter>
        )
    }
}