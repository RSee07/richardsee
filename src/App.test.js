import React from 'react'
import ReactDOM from 'react-dom'
import App from './App'
import { shallow, mount } from 'enzyme'
import toJson from 'enzyme-to-json'

describe('<App /> shallow rendering', () => {
  test('matches snapshot', () => {
    const tree = shallow(<App />)
    expect(toJson(tree)).toMatchSnapshot()
  })
})
