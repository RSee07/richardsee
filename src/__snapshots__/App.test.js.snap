// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<App /> shallow rendering matches snapshot 1`] = `
<BrowserRouter>
  <withRouter(ScrollToTop)>
    <withRouter(Content)
      data={
        Object {
          "careerHistory": Array [
            Object {
              "company": "Spark New Zealand",
              "title": "Online Development Engineer",
              "work": Array [
                Object {
                  "date": "Nov 2017-May 2018",
                  "description": Array [
                    "Built new homepage to deliver personalised customer experiences",
                    "Collaborated with designers and testers to produce highly reusable and robust components to deploy across the Spark website",
                    "Tests written using Jest, Enzyme and Storybook",
                  ],
                  "title": "React Development",
                },
                Object {
                  "date": "Oct 2017-Mar 2018",
                  "description": Array [
                    "Design of captive portals for customers and general public to access free WiFi through phone boxes located nationwide",
                    "Refined connection journey resulting in simpler user experience and text message cost reduction of 40%",
                    "Implemented front-end design to enable WiFi access to Australian and UK mobile numbers",
                    "Worked closely with customers to develop custom captive portals, delivered fast turnaround with high satisfaction",
                  ],
                  "title": "Spark Public WiFi",
                },
              ],
            },
            Object {
              "company": "Spark New Zealand",
              "title": "Associate Development Engineer",
              "work": Array [
                Object {
                  "date": "Sep 2016-Oct 2017",
                  "description": Array [
                    "Built web application to view and send comms for priority incidents using ASP.NET MVC to replace licensed tool purchased annually",
                    "Involved in all stages of software development lifecycle",
                    "Full-stack development - managed database changes using SQL Server right to UI/UX design with jQuery and Kendo UI for webpage interaction",
                    "Completed project resulted in yearly savings of $250,000",
                  ],
                  "title": "ReAssure Communications",
                },
                Object {
                  "date": "Nov 2016-Feb 2017",
                  "description": Array [
                    "Mentored TupuToa intern over summer providing leadership and direction around ReAssure project, reinforced best coding practices",
                    "Day-to-day guidance and introduction to company culture",
                  ],
                  "title": "Mentor",
                },
                Object {
                  "date": "Dec 2015-Jul 2016",
                  "description": Array [
                    "Created an end-to-end environment monitor solution as a side project",
                    "Designed and built a sensor board to feed environmental data into a Raspberry Pi. Data retrieved and aggregated using Python scripts",
                    "Digital Ocean droplet stood up to hold data using InfluxDB, with Splunk and Grafana handling visualisation",
                  ],
                  "title": "Environment Monitor",
                },
              ],
            },
            Object {
              "company": "Cubic Defence New Zealand",
              "title": "Intern",
              "work": Array [
                Object {
                  "date": "Nov 2014-Feb 2015",
                  "description": Array [
                    "Repurposed existing laser detector modules worn by soldiers into a matrix to align weapons more efficiently using C",
                    "Delivered project in sprints implementing the Atlassian Stack",
                    "Hands-on experience with tools and equipment to test and debug",
                  ],
                  "title": "Laser Beam Alignment Tool",
                },
              ],
            },
            Object {
              "company": "Studiosity (formerly YourTutor)",
              "title": "Contractor",
              "work": Array [
                Object {
                  "date": "Feb 2014-Dec 2015",
                  "description": Array [
                    "Tutored Australian students 1:1 online from ages 8-18",
                    "Lessons held through web application connecting students to tutors in real time, with a chat messenger and whiteboard for interaction",
                    "Shared knowledge of subjects, coached and mentored students through questions to achieve better understanding",
                    "Subjects: Maths, Physics, Science, Assignment Research, Study Skills",
                  ],
                  "title": "Online Tutor",
                },
              ],
            },
          ],
          "contact": Object {
            "email": "ricardosee@gmail.com",
            "facebook": "https://www.facebook.com/richardcysee",
            "github": "https://github.com/RSee07",
            "instagram": "https://www.instagram.com/richard_cy_see",
            "linkedIn": "https://www.linkedin.com/in/richard-see",
            "mobile": "+64 21 120 3515",
          },
          "coreSkills": Array [
            "React",
            "HTML",
            "CSS",
            "Javascript",
            "Testing",
            "ASP.NET MVC",
            "SQL",
            "Git",
            "UI/UX design",
            "Agile",
            "Teamwork",
            "Leadership",
            "Effective communication",
            "Critical thinking",
            "Customer focused delivery",
          ],
          "educationAchievements": Array [
            Object {
              "description": "Bachelor of Engineering (Honours) in Computer Systems Engineering",
              "title": "The University of Auckland",
              "year": "2012-2015",
            },
            Object {
              "description": "Parallel and Distributed Systems Category: Distributed Context-Aware Intelligent Environment",
              "title": "Winner of Engineering Part IV Research Project",
              "year": "2015",
            },
            Object {
              "description": "",
              "title": "Wellington College",
              "year": "2007-2011",
            },
          ],
          "interests": Array [
            Object {
              "description": "",
              "title": "Craft Beer",
            },
            Object {
              "description": "Built controller connected to Raspberry Pi to control LIFX lights",
              "title": "Exploring Technologies",
            },
          ],
          "name": "Richard See",
          "profile": Array [
            "I am passionate about software that delivers meaningful benefit to users. I am driven to create customer delight through intuitive digital experiences and to streamline work processes using modern software solutions.",
            "My roles have given me a sound understanding on how to manage ambiguity, and approach issues in a logical manner to reach a fitting outcome. I thrive in a team environment, contributing my own skills in a collaborative approach.",
          ],
        }
      }
    />
  </withRouter(ScrollToTop)>
</BrowserRouter>
`;
